EMAILC-54: IMAP not applying filtering on server side results in larg… (#1093)
mulesoft/docs-connectors: email/1.3/modules/ROOT/pages/email-documentation.adoc

> = Email Connector Reference - Mule 4

Reviewer: ```suggestion
= Email Connector 1.3 Reference - Mule 4
```

:page-aliases: connectors::email/email-documentation.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

> Email Connector v1.3

Reviewer: Remove this line

Anypoint Connector for Email (Email Connector ) is used to list and send emails and perform operations in different mailboxes, such as `delete` and `mark as read`.

Email Connector supports the SMTP, SMTPS, IMAP, IMAPS, POP3, and POP3s protocols.

== Configurations

[[imap]]
=== IMAP

Configuration for operations that are performed through the IMAP (Internet Message Access Protocol) protocol.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| * <<imap_imap, IMAP Connection>>
* <<imap_imaps, IMAPS Connection>>
 | The connection types to provide to this configuration. | | x
| Attachment Naming Strategy a| Enumeration, one of:

** NAME
** NAME_HEADERS
** NAME_HEADERS_SUBJECT |  Enables you to specify a strategy for how to name attachments |  `NAME` |

> | Eagerly Fetch Content a| Boolean |  Indicates whether the retrieved emails should be opened and read. The default value is "true". |  true |

Reviewer: ```suggestion
| Eagerly Fetch Content a| Boolean |  Indicates whether to open and read the retrieved emails  |  true |
```

> | Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when as appropriate. |  |

Reviewer: ```suggestion
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances as appropriate. |  |
```

[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Properties a| Object |  An additional set of custom properties to configure the connection session. |  |
| Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  A time unit that qualifies the `connectionTimeout`, `writeTimeout` and `readTimeout` attributes.  | `SECONDS` |
| Connection Timeout a| Number |  The socket connection timeout value. This attribute works in tandem with `timeoutUnit`.  |  `5` |
| Read Timeout a| Number |  The socket read timeout value. This attribute works in tandem with #timeoutUnit. |  `5` |
| Write Timeout a| Number |  The socket write timeout value. This attribute works in tandem with #timeoutUnit  |  `5` |
| Host a| String |  Host name of the mail server |  | x
| User a| String |  Username used to connect with the mail server |  |
| Password a| String |  Username password to connect with the mail server |  |
| Port a| String |  The port number of the mail server  |  `993` |
| TLS Configuration a| <<Tls>> |  TLS Configuration for the secure connection of the IMAPS protocol |  | x

> | Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  |

Reviewer: ```suggestion
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
```

| Pooling Profile a| <<PoolingProfile>> |  Characteristics of the connection pool |  |
|===

== Operations
* <<delete>>
* <<expungeFolder>>
* <<listImap>>
* <<markAsDeleted>>
* <<markAsRead>>

=== Associated Sources
* <<listener-imap>>

[[pop3]]
=== POP3

Configuration for operations that are performed through the POP3 (Post Office Protocol 3) protocol.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| * <<pop3_pop3, POP3 Connection>>
* <<pop3_pop3s, POP3S Connection>>
 | The connection types to provide to this configuration | | x
| Attachment Naming Strategy a| Enumeration, one of:

** NAME
** NAME_HEADERS
** NAME_HEADERS_SUBJECT |  Enables you to specify a strategy for how to name attachments |  `NAME` |

> | Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when appropriate. |  |

Reviewer: ```suggestion
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances as appropriate. |  |
```

| Pooling Profile a| <<PoolingProfile>> |  Characteristics of the connection pool |  |
|===

=== Supported Operations
* <<listPop3>>

==== Associated Sources
* <<listener-pop3>>

---
[[smtp]]
=== SMTP

Configuration for operations that are performed through the SMTP (Simple Mail Transfer Protocol) protocol.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | The name for this configuration. Connectors reference the configuration with this name. | | x
| Connection a| * <<smtp_smtp, SMTP Connection>>
* <<smtp_smtps, SMTPS Connection>>
 | The connection types to provide to this configuration. | | x
| From a| String |  The "From" address of the message sender |  |
| Default Encoding a| String |  Default character encoding to use in all the messages. If not specified, the default charset in the Mule configuration is used. |  |
| Default Content Transfer Encoding a| String |  |  |

> | Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when appropriate. |  |

Reviewer: ```suggestion
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances as appropriate. |  |
```

[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Properties a| Object |  An additional set of custom properties to configure the connection session |  |
| Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  A TimeUnit which qualifies the `connectionTimeout`, `writeTimeout` and `readTimeout` attributes.  |  `SECONDS` |
| Connection Timeout a| Number |  The socket connection timeout value. This attribute works in tandem with `timeoutUnit`. |  `5` |
| Read Timeout a| Number |  The socket read timeout value. This attribute works in tandem with `timeoutUnit`. |  `5` |
| Write Timeout a| Number |  The socket write timeout value. This attribute works in tandem with `timeoutUnit`. |  `5` |
| Host a| String |  Host name of the mail server. |  | x
| User a| String |  Username used to connect with the mail server. |  |
| Password a| String |  Username password to connect with the mail server. |  |
| Port a| String |  The port number of the mail server. |  `465` |
| TLS Configuration a| <<Tls>> |  TLS Configuration for the secure connection of the SMTPS protocol |  | x

> | Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  |

Reviewer: ```suggestion
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
```

| Pooling Profile a| <<PoolingProfile>> |  Characteristics of the connection pool |  |
|===

=== Supported Operations
* <<send>>

=== Operations

[[delete]]
=== Delete
`<email:delete>`

Deletes the email with the specified email ID from the mailbox.

> For IMAP mailboxes all the messages scheduled for deletion (marked as `DELETED`) are also erased from the folder.

Reviewer: ```suggestion
For IMAP mailboxes, all the messages scheduled for deletion (marked as `DELETED`) are also erased from the folder.
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== For Configurations
* <<imap>>

==== Throws
* EMAIL:ACCESSING_FOLDER
* EMAIL:CONNECTIVITY
* EMAIL:RETRY_EXHAUSTED

[[listImap]]
=== List - IMAP
`<email:list-imap>`

List all the emails (with pagination) in the configured IMAP mailbox folder that matches with the specified IMAP matcher criteria.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Mailbox Folder a| String |  Mailbox folder from which to fetch the emails |  `INBOX` |
| Match with a| <<imap-matcher>> |  Email Matcher that gives the capability of filtering the retrieved emails |  |
| Delete After Retrieve a| Boolean |  Specifies whether to delete the returned emails after they are retrieved |  `false` |
| Page Size a| Number |  Size of the page used by the Paging Provider implementation for fetching the emails from the IMAP server | `10` |

> | Limit a| Number |  Maximum amount of emails retrieved by the operation. Take into account that this limit applies only to the emails effectively retrieved by the operation (the ones that match the IMAPEmailPredicateBuilder criteria) and doesn't imply any restriction over the amount of emails being retrieved from the mailbox server. |  `-1` |

Reviewer: ```suggestion
| Limit a| Number |  Maximum amount of emails retrieved by the operation. Take into account that this limit applies only to the emails effectively retrieved by the operation (the emails that match the IMAPEmailPredicateBuilder criteria) and doesn't imply any restrictions for the number of emails that are retrieved from the mailbox server. |  `-1` |
```

| Streaming Strategy a| * <<repeatable-in-memory-iterable>>
* <<repeatable-file-store-iterable>>

> * non-repeatable-iterable |  Configure to use repeatable streams |  |

Reviewer: ```suggestion
* non-repeatable-iterable |  Configure how Mule processes streams with streaming strategies. Repeatable streams are the default behavior. |  |
```

| Attachment Naming Strategy a| Enumeration, one of:

** NAME
** NAME_HEADERS
** NAME_HEADERS_SUBJECT |  Enables you to specify a strategy for how to name attachments | `NAME` |
| Target Variable a| String |  The name of a variable to store the operation's output |  |

> | Target Value a| String |  An expression to evaluate against the operation's output and store the expression outcome in the target variable |  `#[payload]` |

Reviewer: ```suggestion
| Target Value a| String |  An expression that evaluates the operation's output. The expression outcome is stored in the *Target variable*. |  `#[payload]` |
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Array of Message of [<<StoredEmailContent>>] payload and [<<IMAPEmailAttributes>>] attributes
|===

=== For Configurations
* <<imap>>

==== Throws
* EMAIL:EMAIL_LIST
* EMAIL:CONNECTIVITY

[[markAsDeleted]]
=== Mark As Deleted
`<email:mark-as-deleted>`

> Marks an incoming email as `DELETED`. Emails that are marked for deletion are scheduled for deletion when the folder closes, which means that the email is not physically eliminated from the mailbox folder, but its state changes.  All emails that are marked as `DELETED` are eliminated from the mailbox when one of `IMAPOperations#expungeFolder(MailboxConnection, String)` or `IMAPOperations#delete(MailboxConnection, String, long)` is executed. This operation targets a single email.

Reviewer: ```suggestion
Marks an incoming email as `DELETED`. Emails that are marked for deletion are scheduled for deletion when the folder closes, which means that the email is not physically eliminated from the mailbox folder, but its state changes.  All emails that are marked as `DELETED` are eliminated from the mailbox when `IMAPOperations#expungeFolder(MailboxConnection, String)` or `IMAPOperations#delete(MailboxConnection, String, long)` is executed. This operation targets a single email.
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

=== For Configurations
* <<imap>>

==== Throws
* EMAIL:EMAIL_NOT_FOUND
* EMAIL:ACCESSING_FOLDER
* EMAIL:CONNECTIVITY
* EMAIL:RETRY_EXHAUSTED

[[markAsRead]]
=== Mark As Read
`<email:mark-as-read>`

Marks a single email as `READ` changing its state in the specified mailbox folder. This operation can target a single email.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Mailbox Folder a| String |  Folder where the emails are going to be marked as read | `INBOX` |
| Email ID a| Number |  Email ID Number of the email to mark as read. |  | x
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

=== For Configurations
* <<imap>>

==== Throws
* EMAIL:EMAIL_NOT_FOUND
* EMAIL:ACCESSING_FOLDER
* EMAIL:CONNECTIVITY
* EMAIL:RETRY_EXHAUSTED

[[listPop3]]
=== List - POP3
`<email:list-pop3>`

> List all the emails (with pagination) in the configured POP3 mailbox folder that matches the specified POP3 Matcher criteria. The POP3 protocol does not support the capability of finding specific emails from its UID in a folder to move/delete it, so a parameter `deleteAfterRetrieve` is available for deleting the emails from the server right after being retrieved.

Reviewer: ```suggestion
Lists all the emails (with pagination) in the configured POP3 mailbox folder that matches the specified POP3 Matcher criteria. The POP3 protocol does not support finding a specific email from its UID in a folder to move or delete it, so a `deleteAfterRetrieve` parameter is available for deleting the emails from the server immediately after retrieval.
```

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x

> | Mailbox Folder a| String |  Mailbox folder where the emails are going to be fetched |  INBOX |

Reviewer: ```suggestion
| Mailbox Folder a| String |  Mailbox folder from which to fetch the emails |  INBOX |
```

| Match with a| <<pop3-matcher>> |  Email Matcher that gives the capability of filtering the retrieved emails |  |

> | Delete After Retrieve a| Boolean |  Specifies if the returned emails must be deleted after being retrieved or not. |  `false` |

Reviewer: ```suggestion
| Delete After Retrieve a| Boolean |  Specifies whether to delete returned emails after they are retrieved. |  `false` |
```

| Page Size a| Number |  Size of the page used by the Paging Provider implementation for fetching the emails from the POP3 server |  `10` |

> | Limit a| Number |  Maximum amount of emails retrieved by the operation. Take into account that this limit applies only to the emails effectively retrieved by the operation (the ones that matched the `IMAPEmailPredicateBuilder` criteria) and doesn't imply any restriction over the number of emails being retrieved from the mailbox server. |  -1 |

Reviewer: ```suggestion
| Limit a| Number |  Maximum number of emails retrieved by the operation. Take into account that this limit applies only to the emails effectively retrieved by the operation (the emails that matched the `IMAPEmailPredicateBuilder` criteria) and doesn't imply any restriction over the number of emails that are retrieved from the mailbox server. |  -1 |
```

| Streaming Strategy a| * <<repeatable-in-memory-iterable>>
* <<repeatable-file-store-iterable>>
* non-repeatable-iterable |  Configure to use repeatable streams. |  |
| Attachment Naming Strategy a| Enumeration, one of:

** NAME
** NAME_HEADERS
** NAME_HEADERS_SUBJECT |  Enables you to specify a strategy for how to name attachments | `NAME` |
| Target Variable a| String |  The name of a variable to store the operation's output. |  |

> | Target Value a| String |  An expression to evaluate against the operation's output and store the expression outcome in the target variable |  `#[payload]` |

Reviewer: ```suggestion
| Target Value a| String |  An expression that evaluates the operation's output. The expression outcome is stored in the *Target variable*. |  `#[payload]` |
```

| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors. |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |Array of Message of [<<StoredEmailContent>>] payload and [<<POP3EmailAttributes>>] attributes
|===

=== For Configurations
* <<pop3>>

==== Throws
* EMAIL:EMAIL_LIST
* EMAIL:CONNECTIVITY

[[send]]
=== Send
`<email:send>`

> This operation sends an email message. The message is sent to all recipient  `To`, `CC`, and `BCC` addresses specified in the message.  The content of the message is some type of text (`text/plain`, `text/html`) and is composed of the body and its content type. If no content is specified then the incoming payload is converted into plain text if possible.

Reviewer: ```suggestion
This operation sends an email message. The message is sent to all recipient  (`To`, `CC`, and `BCC`) addresses specified in the message.  The content of the message is some type of text (`text/plain`, `text/html`) and is composed of the body and its content type. If no content is specified then the incoming payload is converted into plain text if possible.
```

| Encoding a| String |  The character encoding of the body. If it is configured, it overrides the one inferred from the body. |  |
| Attachments a| Object |  The attachments that are sent along with the email body |  |
| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that is used to represent the body in an acceptable manner for transport. The value is case insensitive.  Known encodings:  BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY  |  `Base64` |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

=== For Configurations
* <<smtp>>

==== Throws
* EMAIL:SEND
* EMAIL:CONNECTIVITY
* EMAIL:RETRY_EXHAUSTED

== Sources

[[listener-imap]]
=== On New Email - IMAP
`<email:listener-imap>`

Retrieves all the emails from an IMAP mailbox folder, watermark can be enabled for polled items.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Folder a| String |  The name of the folder to poll emails from.  |  `INBOX` |

> | Delete After Retrieve a| Boolean |  Enables the deletion of the polled emails after being retrieved. This is disabled by default. | `false` |

Reviewer: ```suggestion
| Delete After Retrieve a| Boolean |  Enables the deletion of the polled emails after they are retrieved. This is disabled by default. | `false` |
```

> | Enable Watermark a| Boolean |  If watermark should be applied to the polled emails or not. Default to true. |  true |

Reviewer: ```suggestion
| Enable Watermark a| Boolean |  Specifies whether to apply a watermark to the polled emails. |  true |
```

> | Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. By default already read emails will be filtered. |  |

Reviewer: ```suggestion
| Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. By default, already-read emails are filtered. |  |
```

> | Enable Remote Search a| Boolean |  If search filters should be resolved on the server side the value is `true`, or on the client side the value is `false`. Default to `false` because some email servers are not fully compliant with `rfc-3501` search terms. Activating this feature diminishes traffic by reducing the amount of emails brought to the client side for processing. |  `false` |

Reviewer: ```suggestion
| Enable Remote Search a| Boolean |  If search filters should be resolved on the server side, set this value to `true`. The default is `false`, which means search filters are resolved on the client side. This is because some email servers are not fully compliant with `rfc-3501` search terms. Activating this feature diminishes traffic by reducing the number of emails sent to the client side for processing. |  `false` |
```

> | Primary Node Only a| Boolean |  Whether this source should only be executed on the primary node when running in Cluster |  |

Reviewer: ```suggestion
| Primary Node Only a| Boolean |  Whether to execute this source on the primary node only when running in cluster mode |  |
```

| Attachment Naming Strategy a| Enumeration, one of:

** NAME
** NAME_HEADERS
** NAME_HEADERS_SUBJECT |  Enables you to specify a strategy for how to name attachments | `NAME`  |
| Reconnection Strategy a| * <<reconnect>>
* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
|===

==== Output
[%autowidth.spread]
|===
|Type |<<StoredEmailContent>>
| Attributes Type a| <<BaseEmailAttributes>>
|===

=== For Configurations
* <<imap>>

[[listener-pop3]]
=== On New Email - POP3
`<email:listener-pop3>`

Retrieves all the emails from an POP3 mailbox folder.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use | | x
| Folder a| String |  The name of the folder to poll emails from |  `INBOX` |
| Delete After Retrieve a| Boolean |  Enables the deletion of the polled emails after being retrieved. This is disabled by default. |  `false` |
| Pop3 Matcher a| <<pop3-matcher>> |  A matcher to filter emails retrieved by this polling source |  |

> | Primary Node Only a| Boolean |  Whether this source should be executed on the primary node only when running in a cluster |  |

Reviewer: ```suggestion
| Primary Node Only a| Boolean |  Whether to execute this source on the primary node only when running in cluster mode |  |
```

* <<reconnect-forever>> | The reconnection strategy to use |  |
|===

[[reconnect]]
=== Reconnect

[%header%autowidth.spread]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often in milliseconds to reconnect. | |
| Count a| Number | How many reconnection attempts to make. | |
| blocking |Boolean |If false, the reconnection strategy runs in a separate, non-blocking thread. |true |
|===

[[reconnect-forever]]
=== Reconnect Forever

[%header%autowidth.spread]
|===
| Field | Type | Description | Default Value | Required
| Frequency a| Number | How often in milliseconds to reconnect. | |
| blocking |Boolean |If false, the reconnection strategy runs in a separate, non-blocking thread. |true |
|===

[[PoolingProfile]]
=== Pooling Profile

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Max Active a| Number | Controls the maximum number of Mule components that can be borrowed from a session at one time. When set to a negative value, there is no limit to the number of components that can be active at one time. When maxActive is exceeded, the pool is said to be exhausted. |  |

> | Max Idle a| Number | Controls the maximum number of Mule components that can sit idle in the pool at any time. When set to a negative value, there is no limit to the number of Mule components that may be idle at one time. |  |

Reviewer: ```suggestion
| Max Idle a| Number | Controls the maximum number of Mule components that can sit idle in the pool at any time. When set to a negative value, there is no limit to the number of Mule components that can be idle at one time. |  |
```

| Max Wait a| Number | Specifies the number of milliseconds to wait for a pooled component to become available when the pool is exhausted and the exhaustedAction is set to `WHEN_EXHAUSTED_WAIT`. |  |

> | Min Eviction Millis a| Number | Determines the minimum amount of time an object may sit idle in the pool before it is eligible for eviction. When non-positive, no objects will be evicted from the pool due to idle time alone. |  |

Reviewer: ```suggestion
| Min Eviction Millis a| Number | Determines the minimum amount of time an object can sit idle in the pool before it is eligible for eviction. When the value is a negative number, no objects are evicted from the pool due to idle time alone. |  |
```

> | Eviction Check Interval Millis a| Number | Specifies the number of milliseconds between runs of the object evictor. When non-positive, no object evictor is executed. |  |

Reviewer: ```suggestion
| Eviction Check Interval Millis a| Number | Specifies the number of milliseconds between runs of the object evictor. When this value is set to a negative number, no object evictor is executed. |  |
```

| Initialisation Policy a| Enumeration, one of:

** INITIALISE_NONE
** INITIALISE_ONE
** INITIALISE_ALL | Determines how components in a pool should be initialized. The possible values are: `INITIALISE_NONE` (loads no components into the pool on startup), `INITIALISE_ONE` (loads one initial component into the pool on startup), or `INITIALISE_ALL` (loads all components in the pool on startup) |  |
| Disabled a| Boolean | Whether pooling should be disabled |  |
|===

[[Tls]]
=== TlS

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Enabled Protocols a| String | A comma-separated list of protocols enabled for this context |  |
| Enabled Cipher Suites a| String | A comma-separated list of cipher suites enabled for this context |  |
| Trust Store a| <<TrustStore>> |  |  |
| Key Store a| <<KeyStore>> |  |  |
| Revocation Check a| * <<standard-revocation-check>>
* <<custom-ocsp-responder>>
* <<crl-file>> |  |  |
|===

[[TrustStore]]
=== Trust Store

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required

> | Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the trust store |  |

Reviewer: ```suggestion
| Path a| String | The location (which will be resolved relative to the current classpath and file system, if possible) of the truststore |  |
```

Reviewer: I always change trust store to truststore in the text. Same with keystore.

Reviewer: Pero, es no importante.
